Sources/Build/Buildable.swift

```diff
@@ -9,20 +9,43 @@
 */
 
 import PackageType
+import struct Utility.Path
 
 protocol Buildable {
     var targetName: String { get }
     var isTest: Bool { get }
 }
 
+extension CModule {
+    func workingDirectory(prefix: String) -> String {
+        return Path.join(prefix, "\(c99name).build")
+    }
+}
+
 extension Module: Buildable {
     var isTest: Bool {
         return self is TestModule
     }
 
-    var Xcc: [String] {
+    func XccFlagsForPrefix(prefix: String) -> [String] {
         return recursiveDependencies.flatMap { module -> [String] in
-            if let module = module as? CModule {
+            if let module = module as? ClangModule {
+                var moduleMap: String? = nil
+                
+                if module.moduleMapPath.isFile {
+                    moduleMap = module.moduleMapPath
+                }
+    
+                let genModuleMap = Path.join(module.workingDirectory(prefix), module.moduleMap)
+                if genModuleMap.isFile {
+                    moduleMap = genModuleMap
+                }
```

Comment:
This overwrites use of `module.moduleMapPath` if the generated `module.moduleMap` is present, which seems wrong. Also, some comments here explaining intent would be good.

I would prefer that the implementation always create the module map in the build directory, and it do so by either copying the user provided module map or generating one. The reason to always have it in the build directory is it makes downstream commands more consistent, and it more clearly separates out exactly what things are allowed to be referenced. This, however, will require copying all of the headers as well which makes it a more involved project so probably best to defer that to another PR.